### DISCUSSION TITLE
Compatibility with Rails features

### DISCUSSION THREAD

**Discussion by hakimmazouz** [mwnciau/rblade#13] [General]
Something I'm curious on but couldn't find out in the README.

Do all the global functions that work in an ERB template work in an R Blade file?

I'm thinking about things like:
- custom functions from helper files
- vite_client_tag function from vite_ruby
- path helpers

Thank you in advance and sorry fi this was an inappropriate place to ask this, couldn't find anywhere else to do so. And if it's easier to state what doesn't work rather than what does, please enlighten me.

And thank you for this project, I think we really need this approach in Rails land!

**Comment by mwnciau** [comment]
I've just enabled discussions on the repository which seems like it would be a good place for questions like this. I'll move this over now.

**Comment by mwnciau** [comment]
Any global functions that are accessible in an ERB template will be available in an RBlade template. This is more to do with how Rails sets up the template environment than anything this gem does. I have a production project working with RBlade using route helpers, vite and functions from helper files with no problems.

There is one caveat: any helper functions that rely on block content such as simple forms don't quite work yet. I've got an idea on how to get them working, but haven't got around to implementing it yet. To be honest, I'm not a huge fan of the simple form paradigm and would rather move away from that, but I recognise not everyone would agree, and it would make migrating to use RBlade more difficult.

**Comment by mwnciau** [comment]
> There is one caveat: any helper functions that rely on block content such as simple forms don't quite work yet.

As of the 1.2 release, helper functions that rely on block content will now work, so besides some of the more esoteric functionality (e.g. `-%>` etc.), RBlade templates should be able to render ERB templates without any issues.
